documentation/src/main/asciidoc/reference/configuration.asciidoc

```diff
@@ -187,36 +185,110 @@ This includes, but may not be limited to:
+====
+
+[[configuration-bean-reference-parsing]]
+=== Parsing of bean references
+
+When referencing beans using a string value in configuration properties, that string is parsed.
+
+Here are the most common formats:
+
+* `bean:` followed by the name of a Spring or CDI bean.
+For example `bean:myBean`.
+* `class:` followed by the fully-qualified name of a class, to be instantiated through Spring/CDI if available,
+or through its public, no-argument constructor otherwise.
+For example `class:com.mycompany.MyClass`.
+* An arbitrary string that doesn't contain a colon:
+it will be interpreted as explained in <<configuration-bean-resolution>>. In short:
+** first, look for a built-in bean with the given name;
+** then try to retrieve a bean with the given name from Spring/CDI (if available);
+** then try to interpret the string as a fully-qualified class name
+and to retrieve the corresponding bean from Spring/CDI (if available);
+** then try to interpret the string as a fully-qualified class name
+and to instantiate it through its public, no-argument constructor.
+
+The following formats are also accepted, but are only useful for advanced use cases:
+
+* `any:` followed by an arbitrary string.
+Equivalent to leaving out the prefix in most cases. Only useful if the arbitrary string contains a colon.
+* `builtin:` followed by the name of a built-in bean,
+e.g. `simple` for the <<backend-elasticsearch-indexlayout,Elasticsearch index layout strategies>>.
+This will not fall back to Spring/CDI or a direct constructor call.
+* `constructor:` followed by the fully-qualified name of a class, to be instantiated through Spring/CDI if available,
+or through its public, no-argument constructor otherwise.
+This will ignore built-in beans and will not try to instantiate the class through Spring/CDI.
+
 [[configuration-bean-resolution]]
 === Bean resolution
 // Search 5 anchors backward compatibility
 [[section-services]]
 
-Everywhere a custom bean reference is expected, three types of references are accepted:
-
-* A reference by type, as a `java.lang.Class`.
-* A reference by name, as a `java.lang.String`.
-* A reference by type *and* name (through a `BeanReference`, see below).
-
-Bean resolution (i.e. the process of turning this reference into an object instance) happens as follows:
+Bean resolution (i.e. the process of turning this reference into an object instance)
+happens as follows by default:
 
-* If a dependency injection framework is integrated into Hibernate ORM,
+* If the given reference matches a built-in bean, that bean is used.
++
+Example: the name `simple`,
+when used as the value of the property `hibernate.search.backend.layout.strategy`
+to configure the <<backend-elasticsearch-indexlayout,Elasticsearch index layout strategy>>,
+resolves to the built-in `simple` strategy.
+* Otherwise, if a dependency injection framework is integrated into Hibernate ORM,
 the reference is first requested to the DI framework.
```

Comment:
I don't know if `first` still makes here. Even if we are clearly in the `Otherwise` branch, I wouldn't use the term `first`, to be make 100% clear that this check is performed only after we check the `built-in` bean repository.